susan/apps/dhcp.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 <thenakliman@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from susan.db.rdbms import dhcp as dhcp_db
+from susan.lib.dhcp import dhcp
+from susan.lib.dhcp import constants as const
+
+
+# This class provides extra processing needs to be done for requirement
+# specific processing.
+class DHCP(dhcp.DHCPServer):
+    def __init__(self, *args, **kwargs):
+        super(DHCP, self).__init__(*args, **kwargs)
+        self.db = dhcp_db.DHCPDB()
+
+    def handle_discover(self, pkt, datapath, in_port):
+        """Handle discover
+
+           This method provides hooks for extra processing, needs to be done
+           for specific to requirement. If no extra information is needed then
+           call super method.
+        """
+        return super(DHCP, self).handle_discover(pkt, datapath, in_port)
+
+    def handle_request(self, pkt, datapath, in_port):
+        """Handle request
+
+           This method provides hooks for extra processing, needs to be done
+           for specific to requirement. If no extra information is needed then
+           call super method.
+        """
+        return super(DHCP, self).handle_request(pkt, datapath, in_port)
+
+    def get_available_ip(self, datapath, interface):
+        """Gets the free available IP"""
+        return '172.30.10.35'
+
+    def get_subnet_id(self, datapath, interface):
+        return self.db.get_subnet_id(datapath=datapath,
+                                     interface=interface)
+
+    def get_parameters(self, datapath, interface, mac):
```

Comment:
What if subnet does not exist?
What if parameters does not exist?

Check that parameters are returned not the list.